packages/nitro-protocol/src/contract/challenge.ts

```diff
@@ -1,9 +1,50 @@
 import {Bytes32} from './types';
-import {keccak256, defaultAbiCoder} from 'ethers/utils';
-import {State, hashState} from './state';
+import {keccak256, defaultAbiCoder, bigNumberify} from 'ethers/utils';
+import {hashState, VariablePart, State} from './state';
+import {Channel} from '..';
+import {decodeOutcome} from './outcome';
 
 export function hashChallengeMessage(challengeState: State): Bytes32 {
   return keccak256(
     defaultAbiCoder.encode(['bytes32', 'string'], [hashState(challengeState), 'forceMove']),
   );
 }
+
+export interface ChallengeRegisteredEvent {
+  challengerAddress: string;
+  finalizesAt: string;
+  challengeStates: State[];
+}
+export function getChallengeRegisteredEvent(eventArgs): ChallengeRegisteredEvent {
+  // TODO: There must be a better way of parsing out the event
```

Comment:
Based on the [docs](https://docs.ethers.io/ethers.js/html/api-contract.html#event-object) I think that `eventArgs[eventArgs.length - 1]` is an object with the key being the key name in the event.

e.g.,

```
const turnNumRecord = eventArgs[eventArgs.length - 1].turnNumRecord;
```